app/backends/workflow-orchestration.ts

```diff
@@ -153,8 +153,11 @@ export async function updateWorkItem(req: HarmonyRequest, res: Response): Promis
     workItem = await getWorkItemById(tx, parseInt(id, 10));
     const job: Job = await Job.byJobID(tx, workItem.jobID);
 
+    // If the job was already canceled then send 400 response
+    if (job.status === JobStatus.CANCELED) {
+      res.status(400).send();
```

Comment:
I think we send a 409 in other places to indicate a conflict. It's probably best to also have a message indicating the issue.

It would probably be best to throw a `ConflictError` here with a message and then the harmony code at the top level will make sure to format the message with the appropriate content type and status code of 409 (that is you shouldn't have to modify any other code - should just work).